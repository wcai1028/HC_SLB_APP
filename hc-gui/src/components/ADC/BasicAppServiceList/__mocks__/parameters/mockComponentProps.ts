
export default (isLoading: boolean = false) => {
  if (isLoading) {
    return {
      data: new Array,
      isLoading: true,
    }
  } else {
    return {
      data: [
        {
          'app-svc': 'appSvc1',
          vport: '80+tcp',
          'app-svc-group': 'appSvcGrp1',
        },
        {
          'app-svc': 'appSvc2',
          vport: '81+udp',
          'app-svc-group': 'appSvcGrp2',
        },
      ],
      isLoading: false,
    }
  }

}
