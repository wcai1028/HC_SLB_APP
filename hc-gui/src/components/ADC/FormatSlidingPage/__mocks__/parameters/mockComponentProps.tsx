import * as React from 'react'

const minProps = {
  isOpen: true,
  children: <p>Inner Component</p>,
}

const commonProps = {
  isOpen: true,
  title: 'Common Unit Test',
  description: 'This is a common Unit Test',
  children: <p>Inner Component</p>,
  showBack: false,
  saveText: null,
  disableSave: false,
  hideCancel: false,
  customizedButtons: null,
}

const specialProps = {
  isOpen: true,
  title: 'Special Unit Test',
  description: 'This is a special Unit Test',
  children: <p>Inner Component</p>,
  showBack: true,
  saveText: 'TestSave',
  disableSave: true,
  hideCancel: true,
  customizedButtons: <button>Customized</button>,
}

export default (isSpecial?: boolean, callbackOK?: any, callbackCancel?: any) => {
  if (isSpecial === undefined && callbackOK === undefined && callbackCancel === undefined) {
    return minProps
  }

  if (isSpecial) {
    return {
      ...specialProps,
      onRequestOk: callbackOK,
      onRequestClose: callbackCancel,
    }
  } else {
    return {
      ...commonProps,
      onRequestOk: callbackOK,
      onRequestClose: callbackCancel,
    }
  }
}
