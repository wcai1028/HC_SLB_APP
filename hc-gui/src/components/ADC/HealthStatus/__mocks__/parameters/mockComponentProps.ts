export default {
  headerProps: {
    type: 'header',
  },
  ongoingProps: {
    type: 'ongoing',
  },
  lowwarnProps: {
    type: 'lowwarn',
  },
  warningProps: {
    type: 'warning',
  },
  stoppedProps: {
    type: 'stopped',
  },
  infoProps: {
    text: 'test info',
    type: 'info',
    tooltip: 'test tooltip',
  },
  unusedProps: {
    type: 'ununsed',
    default: 'test default'
  },
  numberProps: {
    type: 'number',
    size: 32,
  },
  undefinedProps: {
  },
}
