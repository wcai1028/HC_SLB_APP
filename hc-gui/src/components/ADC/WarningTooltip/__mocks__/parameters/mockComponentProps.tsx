import * as React from 'react'

interface ITestDivProps {
  onClick: () => void,
}

class TestDiv extends React.Component<ITestDivProps> {

  handleClick = () => {
    const { onClick } = this.props
    if (onClick) {
      onClick()
    }
  }

  render() {
    return <div onClick={this.handleClick}>{this.props.children}</div>
  }
}

export const mockWarningTooltipProps =  (useDefault: boolean, onConfirm: any, onVisibleChange: any) => {
  if (useDefault) {
    return {
      title: '',
      description: [],
      action: '',
      onConfirm,
      content: null,
      arrowPointAtCenter: null,
      placement: null,
      trigger: null,
      visible: undefined,
      onVisibleChange: null,
    }
  } else {
    return {
      title: 'Test Title',
      description: ['Test description1', 'Test descriptions2'],
      action: 'Save',
      onConfirm,
      content: (
        title: string,
        description: string[],
        action: string,
        onConfirm: () => void,
      ) => {
        return <>
          <div>{title}</div>
          <div>
            {description.map((data: string, index: number) => {
              return <span key={index}>{data}</span>
            })}
          </div>
          <TestDiv onClick={onConfirm} >{action}</TestDiv>
        </>
      },
      arrowPointAtCenter: false,
      placement: 'bottomRight',
      trigger: 'click',
      visible: false,
      onVisibleChange,
    }
  }
}

export const mockInfoTooltipProps = (isEmpty: boolean, isDefault: boolean, isTest: boolean) => {
  if (isEmpty) {
    return {}
  } else if (isDefault) {
    return {
      objList: [
        { value: 'Test value1' },
        { value: 'Test value2' },
        { value: 'Test value3' },
      ],
    }
  } else {
    if (isTest === true) {
      return {
        objList: [
          { value: 'Test value1' },
          { value: 'Test value2' },
          { value: 'Test value3' },
        ],
        componentDisplay: null,
        textDisplay: 'Test text',
        renderText: (obj: IObject) => {
          return <div>{obj.value}</div>
        },
        actions: 'Test action',
      }
    } else if (isTest === false) {
      return {
        objList: [
          { value: 'Test value1' },
          { value: 'Test value2' },
          { value: 'Test value3' },
        ],
        componentDisplay: <div>Test component</div>,
        textDisplay: '',
        renderText: (obj: IObject) => {
          return <div>{obj.value}</div>
        },
        actions: 'Test action',
      }
    }
    return {}
  }
}
