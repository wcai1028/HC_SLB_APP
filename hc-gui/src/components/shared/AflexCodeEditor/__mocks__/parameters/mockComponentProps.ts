
export default (onAFlexChange: any, hasDiff: boolean = false, isBeforeStyle: boolean = false) => {
  if (hasDiff) {
    return {
      onAFlexChange,
      value: 'test-value',
      options: {
        readOnly: true,
      },
      diff: [
        ['replace', 1, 3, 5, 7],
        ['insert', 1, 3, 5, 7],
        ['delete', 1, 3, 5, 7],
        ['others', 1, 3, 5, 7],
      ],
      version: isBeforeStyle ? 'before' : 'after',
      id: '123456',
    }
  } else {
    return {
      onAFlexChange,
      value: 'test-value',
      options: {
        readOnly: true,
      },
      diff: null,
      version: 'unit_test',
      id: '123456',
    }
  }

}
