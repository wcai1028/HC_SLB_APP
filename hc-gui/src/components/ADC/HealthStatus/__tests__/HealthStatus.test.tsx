import * as React from 'react'
import * as ReactTestRenderer from 'react-test-renderer'

import mockComponentProps from '../__mocks__/parameters/mockComponentProps'
const {
  headerProps,
  ongoingProps,
  lowwarnProps,
  warningProps,
  stoppedProps,
  infoProps,
  unusedProps,
  numberProps,
  undefinedProps,
} = mockComponentProps

const renderTestComponent = (props: object) => {
  const HCProvider = require('src/HCProvider').default
  const HealthStatus = require('../index').default

  return ReactTestRenderer.create(
    <HCProvider>
      <HealthStatus {...props} />
    </HCProvider>
  )
}

describe('Unit Test: HealthStatus ', () => {

  beforeEach(() => {
    jest.doMock('a10-gui-widgets', () => {
      const guiWidgets = require.requireActual('a10-gui-widgets')
      guiWidgets.A10Tooltip = (props: any) => {
        return (
          <div className="mock-a10-tooltip">
            <div className="mock-tooltip">{props.title ? props.title : null}</div>
            {props.children}
          </div>
        )
      }
      return guiWidgets
    })
  })

  afterEach(() => {
    jest.dontMock('a10-gui-widgets')
  })

  describe('should render properly: ', () => {
    test('should render properly with header type', (done: any) => {
      const testComponent = renderTestComponent(headerProps)

      setImmediate(() => {
        expect(testComponent.toJSON()).toMatchSnapshot()
        done()
      })
    })

    test('should render properly with ongoing type', (done: any) => {
      const testComponent = renderTestComponent(ongoingProps)

      setImmediate(() => {
        expect(testComponent.toJSON()).toMatchSnapshot()
        done()
      })
    })

    test('should render properly with lowwarn type', (done: any) => {
      const testComponent = renderTestComponent(lowwarnProps)

      setImmediate(() => {
        expect(testComponent.toJSON()).toMatchSnapshot()
        done()
      })
    })

    test('should render properly with warning type', (done: any) => {
      const testComponent = renderTestComponent(warningProps)

      setImmediate(() => {
        expect(testComponent.toJSON()).toMatchSnapshot()
        done()
      })
    })

    test('should render properly with stopped type', (done: any) => {
      const testComponent = renderTestComponent(stoppedProps)

      setImmediate(() => {
        expect(testComponent.toJSON()).toMatchSnapshot()
        done()
      })
    })

    test('should render properly with info type', (done: any) => {
      const testComponent = renderTestComponent(infoProps)

      setImmediate(() => {
        expect(testComponent.toJSON()).toMatchSnapshot()
        done()
      })
    })

    test('should render properly with unused type', (done: any) => {
      const testComponent = renderTestComponent(unusedProps)

      setImmediate(() => {
        expect(testComponent.toJSON()).toMatchSnapshot()
        done()
      })
    })

    test('should render properly with number type', (done: any) => {
      const testComponent = renderTestComponent(numberProps)

      setImmediate(() => {
        expect(testComponent.toJSON()).toMatchSnapshot()
        done()
      })
    })

    test('should render properly with undefined type', (done: any) => {
      const testComponent = renderTestComponent(undefinedProps)

      setImmediate(() => {
        expect(testComponent.toJSON()).toMatchSnapshot()
        done()
      })
    })

  })

  describe('should work properly: ', () => {
    test('should update properly when props change', (done: any) => {
      const testComponent = renderTestComponent(ongoingProps)

      const instance = testComponent.root.find(
        (node: any) => (node.type as any).name === 'HealthStatus',
      )
      instance.instance.componentWillReceiveProps(stoppedProps)

      setImmediate(() => {
        expect(testComponent.toJSON()).toMatchSnapshot()
        done()
      })
    })
  })

})
