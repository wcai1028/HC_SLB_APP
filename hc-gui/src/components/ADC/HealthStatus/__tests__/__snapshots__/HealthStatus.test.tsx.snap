// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test: HealthStatus  should render properly:  should render properly with header type 1`] = `
<div
  className="mock-a10-tooltip"
>
  <div
    className="mock-tooltip"
  >
    header
  </div>
  <span
    className="statusCircleBg size16 status-header"
  >
    <span
      className="text"
    >
      
    </span>
  </span>
</div>
`;

exports[`Unit Test: HealthStatus  should render properly:  should render properly with info type 1`] = `
<div
  className="mock-a10-tooltip"
>
  <div
    className="mock-tooltip"
  >
    test tooltip
  </div>
  <span
    className="statusCircleBg size16 status-info"
  >
    <span
      className="text"
    >
      test info
    </span>
  </span>
</div>
`;

exports[`Unit Test: HealthStatus  should render properly:  should render properly with lowwarn type 1`] = `
<div
  className="mock-a10-tooltip"
>
  <div
    className="mock-tooltip"
  >
    lowwarn
  </div>
  <span
    className="statusCircleBg size16 status-lowwarn"
  >
    <span
      className="text"
    >
      
    </span>
  </span>
</div>
`;

exports[`Unit Test: HealthStatus  should render properly:  should render properly with number type 1`] = `
<div
  className="mock-a10-tooltip"
>
  <div
    className="mock-tooltip"
  >
    0
  </div>
  <span
    className="statusCircleBg size32 status-number"
  >
    <span
      className="text"
    >
      0
    </span>
  </span>
</div>
`;

exports[`Unit Test: HealthStatus  should render properly:  should render properly with ongoing type 1`] = `
<div
  className="mock-a10-tooltip"
>
  <div
    className="mock-tooltip"
  >
    ongoing
  </div>
  <span
    className="statusCircleBg size16 status-ongoing"
  >
    <span
      className="text"
    >
      
    </span>
  </span>
</div>
`;

exports[`Unit Test: HealthStatus  should render properly:  should render properly with stopped type 1`] = `
<div
  className="mock-a10-tooltip"
>
  <div
    className="mock-tooltip"
  >
    stopped
  </div>
  <span
    className="statusCircleBg size16 status-stopped"
  >
    <span
      className="text"
    >
      
    </span>
  </span>
</div>
`;

exports[`Unit Test: HealthStatus  should render properly:  should render properly with undefined type 1`] = `
<div
  className="mock-a10-tooltip"
>
  <div
    className="mock-tooltip"
  >
    undefined
  </div>
  <span
    className="statusCircleBg size16 status-undefined"
  >
    <span
      className="text"
    >
      
    </span>
  </span>
</div>
`;

exports[`Unit Test: HealthStatus  should render properly:  should render properly with unused type 1`] = `
<div
  className="mock-a10-tooltip"
>
  <div
    className="mock-tooltip"
  >
    ununsed
  </div>
  <span
    className="statusCircleBg size16 status-ununsed"
  >
    <span
      className="text"
    >
      test default
    </span>
  </span>
</div>
`;

exports[`Unit Test: HealthStatus  should render properly:  should render properly with warning type 1`] = `
<div
  className="mock-a10-tooltip"
>
  <div
    className="mock-tooltip"
  >
    warning
  </div>
  <span
    className="statusCircleBg size16 status-warning"
  >
    <span
      className="text"
    >
      
    </span>
  </span>
</div>
`;

exports[`Unit Test: HealthStatus  should work properly:  should update properly when props change 1`] = `
<div
  className="mock-a10-tooltip"
>
  <div
    className="mock-tooltip"
  >
    stopped
  </div>
  <span
    className="statusCircleBg size16 status-stopped"
  >
    <span
      className="text"
    >
      
    </span>
  </span>
</div>
`;
